torch/cuda/__init__.py

```diff
@@ -1137,6 +1137,46 @@ def _dtype(self):
 torch._storage_classes.add(ComplexDoubleStorage)
 torch._storage_classes.add(ComplexFloatStorage)
 
+
+class _WrappedTritonKernel(object):
+    """ Just a simple wrapper to store some metadata for testing purposes.
+    """
+
+    def __init__(self, kernel):
+        self.kernel = kernel
+        self.kernel_invoked = False
+
+    def __call__(self, *args, **kwargs):
+        res = self.kernel(*args, **kwargs)
+        self.kernel_invoked = True
+        return res
+
+
```

Comment:
Don't you plan to remove it in the final version of PR?

Reply:
Do we have to? Otherwise how could I test that the dispatch in `nn.Linear` actually worked?

Reply:
But otherwise, this introduces overhead in the actual usecases of the the kernel, outside of test, isn't it?